sdk/core/Azure.Core.Experimental/src/Variant/Variant.cs

```diff
@@ -1388,5 +1388,41 @@ private Variant(object o, ulong u)
             return value;
         }
         #endregion
+
+        /// <inheritdoc/>
+        public override string? ToString()
+        {
+            string? result;
+
+            if (_object == null)
+            {
+                result = "null";
+            }
+            else if (_object is TypeFlag flag)
+            {
+                if (Type == typeof(bool))
+                {
+                    result = flag.ToObject(this).ToString()!.ToLowerInvariant();
+                }
+                else
+                {
+                    result = flag.ToObject(this).ToString();
```

Comment:
this should also be optimized for common primitives. Feel free to improve it in a subsequent release.

Reply:
Are you thinking we would add a switch statement on Type and use that to not box the known value-type primitives?

Reply:
Done.